interfaces/acados_template/acados_template/acados_sim.py

```diff
@@ -151,27 +189,66 @@ def T(self, T):
 
     @num_stages.setter
     def num_stages(self, num_stages):
-
-        if type(num_stages) == int:
+        if isinstance(num_stages, int):
             self.__sim_method_num_stages = num_stages
         else:
-            raise Exception('Invalid num_stages value. num_stages must be an integer. Exiting.')
+            raise Exception('Invalid num_stages value. num_stages must be an integer.')
 
     @num_steps.setter
     def num_steps(self, num_steps):
-
-        if type(num_steps) == int:
+        if isinstance(num_steps, int):
             self.__sim_method_num_steps = num_steps
         else:
-            raise Exception('Invalid num_steps value. num_steps must be an integer. Exiting.')
+            raise Exception('Invalid num_steps value. num_steps must be an integer.')
 
     @newton_iter.setter
     def newton_iter(self, newton_iter):
-
-        if type(newton_iter) == int:
+        if isinstance(newton_iter, int):
             self.__sim_method_newton_iter = newton_iter
         else:
-            raise Exception('Invalid newton_iter value. newton_iter must be an integer. Exiting.')
+            raise Exception('Invalid newton_iter value. newton_iter must be an integer.')
+
+    @sens_forw.setter
+    def sens_forw(self, sens_forw):
+        if sens_forw in (True, False):
```

Comment:
very minor, but why not `isinstance(sens_forw, bool)`?

Reply:
It is obviously equivalent.
There are some checks like: 
```
integrator_types = ('ERK', 'IRK', 'GNSF')
if type(integrator_type) == str and integrator_type in integrator_types:
```
where I removed the type check, since it is also covered with the second check.
I guess it propagated from adapting this.